docs/pages/access-controls/reference.mdx

```diff
@@ -373,3 +374,188 @@ Here is an explanation of the fields used in the `where` and `filter` conditions
+
+#### `{{internal.kubernetes_groups}}`
+
+- **Type**: Variable
+- **Description**: List of allowed Kubernetes groups.
+- **Example**: `{{internal.kubernetes_groups}}` -> `["group1", "group2"]`
+
+#### `{{internal.kubernetes_users}}`
+
+- **Type**: Variable
+- **Description**: List of allowed Kubernetes users.
+- **Example**: `{{internal.kubernetes_users}}` -> `["user1", "user2"]`
+
+#### `{{internal.logins}}`
+
+- **Type**: Variable
+- **Description**: List of allowed SSH logins.
+- **Example**: `{{internal.logins}}` -> `["user1", "user2"]`
+
+#### `{{internal.windows_logins}}`
+
+- **Type**: Variable
+- **Description**: List of allowed Windows logins.
+- **Example**: `{{internal.windows_logins}}` -> `["user1", "user2"]`
+
+### External Variables
+
+These variables provide context from an Identity Provider or other external data source. We can't predict every variable an external data source will provide, but these are common values expected from SAML and OIDC identity providers.
+
+#### `{{external.logins}}`
+
+- **Type**: Variable
+- **Provider**: Common across several IdPs
+
+#### `{{external.username}}`
+
+- **Type**: Variable
+- **Provider**: Common across several IdPs
+
+#### `{{external.access["label"]}}`
+
+- **Type**: Variable
+- **Provider**: Common across several IdPs
+
+#### `{{external.env}}`
+
+- **Type**: Variable
+- **Provider**: Common across several IdPs
+
+#### `{{external.first_name}}`
+
+- **Type**: Variable
+- **Description**: Fetches the user's first name from the SSO provider.
+- **Example**: `{{external.first_name}}` -> `"Alice"`
+- **Provider**: OIDC/SAML
+
+**Note**: Per OIDC specifications this value may be given as `given_name`.
+
+#### `{{external.last_name}}`
+
+- **Type**: Variable
+- **Description**: Fetches the user's last name from the SSO provider.
+- **Example**: `{{external.last_name}}` -> `"Smith"`
+- **Provider**: OIDC/SAML
+
+**Note**: Per OIDC specifications this value may be given as `family_name`.
+
+#### `{{external.email}}`
+
+- **Type**: Variable
+- **Description**: Fetches the user's email address from the SSO provider.
+- **Example**: `{{external.email}}` -> `"alice@example.com"`
+- **Provider**: OIDC/SAML
+
+#### `{{external.groups}}`
+
+- **Type**: Variable
+- **Description**: Fetches the user's group membership from the SSO provider.
+- **Example**: `{{external.groups}}` -> `["group1", "group2"]`
+- **Provider**: OIDC/SAML
+
+#### `{{external.azure_identities}}`
+
+- **Type**: Variable
+- **Description**: Fetches the user's Azure identities from the SSO provider [(2)].
+- **Example**: `{{external.azure_identities}}` -> `["identity1"]`
+- **Provider**: OIDC/SAML
+
+#### `{{external.databases}}`
+
+- **Type**: Variable
+- **Description**: Fetches the user's allowed database names from the SSO provider [(3)].
+- **Example**: `{{external.databases}}` -> `["db1", "db2"]`
+- **Provider**: OIDC/SAML
+
+#### `{{external["custom_attribute"]}}`
```

Comment:
I think it would be nice to document that `external.trait_name` is just a shorthand for `external["trait_name"]`, and the latter is necessary when `trait_name` includes certain special characters like space or hyphen. Same is true for `internal`. Maybe we would even just switch to only documenting the `[]` syntax? I'm not sure

Reply:
Sounds a lot like what's on L557, but could maybe use some tweaking.